mock/src/test_ctx.rs

```diff
@@ -142,12 +142,25 @@ impl<const NACC: usize, const NTX: usize> TestContext<NACC, NTX> {
             .skip(1)
             .for_each(|(idx, tx)| {
                 let idx = u64::try_from(idx).expect("Unexpected idx conversion error");
-                tx.transaction_idx(idx).nonce(idx);
+                tx.transaction_idx(idx);
             });
         let tx_refs = transactions.iter_mut().collect();
 
         // Build Tx modifiers.
         func_tx(tx_refs, accounts.clone());
+
```

Comment:
- Since using invalid nonce is not useful, I've just made the functions `pub(crate)`. 
- The tx_idx logic is moved to after `func_tx`. 
- Also rebased my branch over the latest main.